### PR TITLE
Add new methods to artifact interface: Read, write, parse url

### PR DESCRIPTION
**Pull request by ariefrahmansyah**
Add new methods to the artifact.Service interface to be used by Merlin's image builder improvement: https://github.com/caraml-dev/merlin/pull/500

New methods:

1. ReadArtifact
2. WriteArtifact
3. ParseURL which replace getGcsBucketAndLocation

### CODE REVIEW

File: api/pkg/artifact/artifact.go [hunk 1]
```diff
@@ -2,31 +2,121 @@ package artifact
...
+	// Object is the name and or path of the object stored in the bucket. It
+	// should not start with a forward slash.
+	Object string
+}
+
 type Service interface {
+	ParseURL(gsURL string) (*URL, error)
+
+	ReadArtifact(ctx context.Context, url string) ([]byte, error)
+	WriteArtifact(ctx context.Context, url, content string) error
```
Comment (krithika369) [inline api/pkg/artifact/artifact.go:30]: Should we make `content` here `[]byte` for consistency with the read method (which is also more generic than string)?

File: api/pkg/artifact/artifact.go [hunk 2]
```diff
@@ -2,31 +2,121 @@ package artifact
...
+	u, err := gac.ParseURL(url)
+	if err != nil {
+		return nil, err
+	}
+
+	reader, err := gac.API.Bucket(u.Bucket).Object(u.Object).NewReader(ctx)
+	if err != nil {
+		return nil, err
+	}
+	defer reader.Close() //nolint:errcheck
```
Comment (krithika369) [inline api/pkg/artifact/artifact.go:82]: Why is the `//nolint:errcheck` added here?

**Review by krithika369 — APPROVED** [review]
Left some small comments. The rest LGTM. Thanks, Arief!

File: api/pkg/artifact/artifact.go (hunk 1 above)
Comment (ariefrahmansyah) [reply api/pkg/artifact/artifact.go]: Good idea. Updated

File: api/pkg/artifact/artifact.go (hunk 2 above)
Comment (ariefrahmansyah) [reply api/pkg/artifact/artifact.go:82]: Because reader.Close() returns error and since we're not checking the value of error, linter will fail there.

**Pull request merged**